mapie/quantile_regression.py

```diff
@@ -0,0 +1,323 @@
+from __future__ import annotations
+import warnings
+from typing import List, Optional, Tuple, Union, cast
+
+import numpy as np
+from sklearn.base import RegressorMixin, clone
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import QuantileRegressor
+from sklearn.model_selection import BaseCrossValidator
+from sklearn.utils.validation import (
+    indexable,
+    check_is_fitted,
+    # _num_samples,
+    _check_y,
+)
+
+from ._typing import ArrayLike, NDArray
+from .utils import (
+    check_alpha,
+    check_alpha_and_n_samples,
+    check_n_features_in,
+    # check_n_jobs,
+    # check_nan_in_aposteriori_prediction, <-- check what happens
+    # when you also include this!
+    check_null_weight,
+    # check_verbose,
+    fit_estimator,
+)
+from ._compatibility import np_quantile
+from .regression import MapieRegressor
+
+
+class MapieQuantileRegressor(MapieRegressor):
+    valid_methods_ = ["quantile"]
+    valid_agg_functions_ = [None, "median", "mean"]
+    fit_attributes = [
+        "list_estimators",
+        "list_y_preds_calib",
+        "conformity_scores_",
+        "y_calib"
+    ]
+
+    link_estimator_quantile = {
+        "GradientBoostingRegressor": ["loss", "alpha"],
+        "QuantileRegressor": ["quantile", "quantile"]
+    }
+
+    def __init__(
+        self,
+        estimator: RegressorMixin = None,
+        method: str = "quantile",
+        cv: Optional[Union[int, str, BaseCrossValidator]] = None,
+        n_jobs: Optional[int] = None,
+        agg_function: Optional[str] = "mean",
+        verbose: int = 0,
+        alpha: float = 0.2
+    ) -> None:
+        self.alpha = alpha
+        super().__init__(
+            estimator=estimator,
+            method=method,
+            cv=cv,
+            n_jobs=n_jobs,
+            agg_function=agg_function,
+            verbose=verbose,
+        )
+        if n_jobs is not None:
+            raise NotImplementedError
+
+    def _check_alpha(
+        self,
+        alpha: float,
+    ) -> ArrayLike:
+        if isinstance(alpha, float) is False:
```

Comment:
Following `_check_alpha` in utils, I would simplify it to sth like :
```
    if alpha is None:
        return 0.2
    if isinstance(alpha, float):
        alpha_np = np.array([alpha / 2, 1 - alpha /2 ])  # add 0.5 here ?
    else:
        raise ValueError(
            "Invalid alpha. Allowed values are float."
        )

Reply:
Agreed, implemented together with the method mentionned in the comment afterwards.